middlewares/metrics_test.go

```diff
@@ -3,18 +3,22 @@ package middlewares
 import (
 	"testing"
 
+	"net/http"
+	"net/http/httptest"
+
 	"github.com/go-kit/kit/metrics"
 )
 
 func TestMetricsRetryListener(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
 	// nil implementation, nothing should fail
 	retryListener := NewMetricsRetryListener(nil)
-	retryListener.Retried(1)
+	retryListener.Retried(req, 1)
 
 	retryMetrics := newCollectingMetrics()
 	retryListener = NewMetricsRetryListener(retryMetrics)
-	retryListener.Retried(1)
-	retryListener.Retried(2)
+	retryListener.Retried(req, 1)
+	retryListener.Retried(req, 2)
```

Comment:
Shouldn't we also somehow expand our test expectation / assertion logic when we extend the system-under-test?